pennylane/shadows/transforms.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classical shadow transforms"""
+
+from itertools import product
+from functools import reduce, wraps
+import warnings
+
+import pennylane as qml
+import pennylane.numpy as np
+
+
+@qml.batch_transform
+def __replace_obs(tape, obs, *args, **kwargs):
+    """
+    Tape transform to replace the measurement processes with the given one
+    """
+    with qml.tape.QuantumTape() as new_tape:
+        # queue everything from the old tape except the measurement processes
+        for op in tape.operations:
+            qml.apply(op)
+
+        # queue the new observable
+        obs(*args, **kwargs)
+
+    def processing_fn(res):
+        return qml.math.squeeze(qml.math.stack(res))
```

Comment:
Why not return `res` directly?

Reply:
As with all device code, `res` may have an extra leading dimension.